accepted/0003-plugin-manager.md

```diff
@@ -0,0 +1,114 @@
+- Start Date: 2019-12-30
+- RFC PR: #4
+- Mantis Issue: N/A
+
+# Summary
+Create a plugin manager that is capable of configuring the loading of plugins as
+well as providing an easier means of installing plugins. Perhaps - later down
+the line - creating a online database that individuals can search and seamlessly
+install plugins.
+
+# Motivation
+Varying from plugin to plugin, there are a variety of different intallation
+methods for each plugin. On top of that, once installed, there is no easy way to
+disable, or even configure, each individual plugin. In addition, if one plugin
+doesn't behave correctly because of another, it can become difficult to diagnose
+which plugins are causing the error. By implementing a plugin manager, it allows
+users to seamlessly install plugins and have finer control over the loading of
+their plugins. Furthermore, having a plugin manager opens up many more
+possibilties for both users and developers.
+
+# Design
+## Functionality
+### Loading
+Currently, when plugins cause issues, the only way to disable them is by
+outright uninstalling them. However, this could be mitigated at startup by
+having an internal record of which plugins are installed. By keeping a list -
+perhaps in the configuration file - of the disabled plugins, they could be
+ignored during initilizaiton.
+
+Furthermore, if a plugin causes a major failure while loading or doing use, it
+will be automatically disabled on the next startup with a popup warning the
+user. The popup will allow the user to re-enable the plugin (perhaps they
+knowingly used a experimental feature, etc.) or keep the plugin disabled. It
+would also be helpful to show an error code to the user to help them and the
+developer debug.
+
+### Installation
+To further ease the user with their plugin (once downloaded), installation could be handled in 3
+different ways.
+- A file explorer in the manager to install a compressed package that contains:
+  a manifest, locales, the actual plugin, and other needed components.
```

Comment:
Since that comment we have obviously merged lots of changes on obs-studio, including the bundle-based plugin format on macOS, which bundles (hence the name) everything a plugin consists of into a single package (albeit it just being a directory on macOS, it is treated as a self-contained package by the OS), which holds all binaries, support files, and resources that make up a plugin.

Removing a plugin just requires removing the package (whether manually or by a plugin manager). A similar pattern would probably benefit Windows and Linux users as well.